Wrong documentation URL in error message

<!--- 
Please follow the instructions below. 
We receive dozens of issues every week, so to stay productive, we will close issues that don't provide enough information. 

Please open Android-related issues on the Android Issue Tracker at https://source.android.com/source/report-bugs
Please open Gradle Native-related issues at https://github.com/gradle/gradle-native/issues
-->

<!--- Provide a brief summary of the issue in the title above -->

### Expected Behavior
If a plugin cannot be found, the error message should refer to the list of core plugins at https://docs.gradle.org/7.5.1/userguide/plugin_reference.html.

### Current Behavior
The error message refers to https://docs.gradle.org/7.5.1/userguide/standard_plugins.html, which does not exist. 

### Context
<!--- How has this issue affected you? What are you trying to accomplish? -->
<!--- Providing context helps us come up with a solution that is most useful in the real world -->
I used a wrong plugin ID and tried to follow the reference to the documentation to look up the correct one.

### Steps to Reproduce 
1. Create a `build.gradle` that contains `plugins { id "unknown" }`, no other files are needed
2. Run `gradle`

You will get this error message:
<pre>
FAILURE: Build failed with an exception.

* Where:
Build file 'C:\Tmp\build.gradle' line: 1

* What went wrong:
Plugin [id: 'unknown'] was not found in any of the following sources:

- Gradle Core Plugins (not a core plugin, please see https://docs.gradle.org/7.4.2/userguide/standard_plugins.html for available core plugins)
- Plugin Repositories (plugin dependency must include a version number for this source)

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org
</pre>
<!---
Provide a self-contained example project as an attached archive or a Github project.
You can use the following template with a Gradle GitHub action set up to showcase your problem: https://github.com/gradle/gradle-issue-reproducer
In the rare cases where this is infeasible, we will also accept a detailed set of instructions.
-->

### Your Environment
Gradle 6.8.1, 7.3.3, or 7.5.1 refer all to the non-existing page

--- a/subprojects/plugin-use/src/main/java/org/gradle/plugin/use/resolve/internal/CorePluginResolver.java
+++ b/subprojects/plugin-use/src/main/java/org/gradle/plugin/use/resolve/internal/CorePluginResolver.java
@@ -46,7 +46,7 @@ public class CorePluginResolver implements PluginResolver {
 
         PluginImplementation<?> plugin = pluginRegistry.lookup(id);
         if (plugin == null) {
-            result.notFound(getDescription(), format("not a core plugin, please see %s for available core plugins", documentationRegistry.getDocumentationFor("standard_plugins")));
+            result.notFound(getDescription(), format("not a core plugin, please see %s for available core plugins", documentationRegistry.getDocumentationFor("plugin_reference")));
             return;
         }
 

--- a/subprojects/plugin-use/src/test/groovy/org/gradle/plugin/use/resolve/internal/CorePluginResolverTest.groovy
+++ b/subprojects/plugin-use/src/test/groovy/org/gradle/plugin/use/resolve/internal/CorePluginResolverTest.groovy
@@ -37,7 +37,9 @@ class CorePluginResolverTest extends Specification {
         }
     }
 
-    def docRegistry = Mock(DocumentationRegistry)
+    def docRegistry = Mock(DocumentationRegistry) {
+        it.getDocumentationFor(_ as String) >> { String page -> "https://docs.gradle.org/current/userguide/${page}.html"}
+    }
     def pluginRegistry = Mock(PluginRegistry)
     def result = Mock(PluginResolutionResult)
 
@@ -101,7 +103,9 @@ class CorePluginResolverTest extends Specification {
 
         then:
         1 * pluginRegistry.lookup(DefaultPluginId.of("org.gradle.foo")) >> null
-        1 * result.notFound(resolver.getDescription(), { it.contains("not a core plugin") })
+        1 * result.notFound(resolver.getDescription(), {
+            it.contains("not a core plugin, please see https://docs.gradle.org/current/userguide/plugin_reference.html for available core plugins")
+        })
     }
 
 }

